x-pack/plugins/ingest_manager/server/services/epm/registry/index.ts

```diff
@@ -35,6 +35,19 @@ export interface CategoriesParams {
   experimental?: boolean;
 }
 
+/**
+ * Extract the package name and package version from a string.
+ *
+ * @param pkgkey a string containing the package name delimited by the package version
+ */
+export function splitPkgKey(pkgkey: string): { pkgName: string; pkgVersion: string } {
+  // this will return an empty string if `indexOf` returns -1
+  const pkgName = pkgkey.substr(0, pkgkey.indexOf('-'));
+  // this will return the entire string if `indexOf` return -1
+  const pkgVersion = pkgkey.substr(pkgkey.indexOf('-') + 1);
+  return { pkgName, pkgVersion };
```

Comment:
I don't think it should accept an empty name or version.

I also think we should ensure the version is valid semver  https://github.com/npm/node-semver#usage

Reply:
Thanks for the comment @jfsiii, how about if I check if the `pkgName` is an empty string and throw an error?

For the `pkgVersion` I can do the same thing, `semver.valid(pkgVersion)` and if not throw and error?